src/hypertable.c

```diff
@@ -2461,14 +2461,23 @@ ts_chunk_get_osm_slice_and_lock(int32 osm_chunk_id, int32 time_dim_id, LockTuple
 				.lockmode = tuplockmode,
 				.waitpolicy = LockWaitBlock,
 			};
+			/*
+			 * We cannot acquire a tuple lock when running in recovery mode
```

Comment:
```suggestion
			 it is not possible to acquire tuple lock when running in recovery mode as they require a transaction id and transaction ids cannot be created  in recovery mode, so we only acquire the tuple lock if we are not in recovery mode.
```

Reply:
Added some more detail to the comment to explain why tuple locks cannot be acquired in recovery mode.

Reply:
Added a clarification